lib/generators/json_serializable_generator.dart

```diff
@@ -48,7 +50,8 @@ class JsonSerializableGenerator
     var buffer = new StringBuffer();
 
     if (annotation.createFactory) {
-      var toSkip = _writeFactory(buffer, classElement, fields, prefix);
+      var toSkip =
+          writeFactory(buffer, classElement, fields, prefix, typeHelpers);
```

Comment:
nit: Imo `writeFactory` should actually be a private method in this class (and all the other ones like it). They are only used by this class afaict and you could remove the `typeHelpers` arg that way.